R/Prediction.R

```diff
@@ -138,7 +138,16 @@ makePrediction.SurvTaskDesc = function(task.desc, row.names, id, truth, predict.
   # FIXME: recode times
   data$truth.time = truth[, 1L]
   data$truth.event = truth[, 2L]
-  data$response = y
+  if (predict.type == "response") {
+    data$response = y
+  } else {
+    data$response = y$preds
+    # Extend probs to the truth time points
+    all.times = sort(unique(c(truth[, 1L], y$train.times)))
+    prob_columns = sapply(all.times, function(t) max(which(y$train.times <= t)))
```

Comment:
Please replace `sapply` by the appropriate *apply that doesn't do automatic conversions (lapply?).

Reply:
+1
You might want to use `viapply()`?

Reply:
ok